fineract-provider/dependencies.gradle

```diff
@@ -80,7 +80,8 @@ dependencies {
             'org.webjars:webjars-locator-core',
 
             'com.google.cloud.sql:mysql-socket-factory-connector-j-8:1.2.1',
-            'com.squareup.retrofit2:converter-gson'
+            'com.squareup.retrofit2:converter-gson',
+            'com.sun.activation:jakarta.activation:1.2.2'
```

Comment:
In addition to adding the implementation jar here, you need to add an exclusion for the API that is brought in by an implied dependency from jakarta-xml.bind-api. Otherwise you get duplicates for the API classes.

Reply:
Great!! . I was in a dilemma on how that duplicate jakarta-api dependency gets installed. I'll try that. First I'll resolve this failing test cases.

Reply:
> Great!! . I was in a dilemma on how that duplicate jakarta-api dependency gets installed. I'll try that. First I'll resolve this failing test cases.

@BLasan  i excluded and it worked see this https://github.com/apache/fineract/pull/1627/commits/cff63783cd77beb92cb92e5449ad77002e4ac6ad

Reply:
Yuh. Your approach is ok @francisguchie . But instead of excluding that path, @ptuomola suggested to an exclusion for this particular dependency. I tried by adding it. But didn't work.

```
    implementation ('com.sun.activation:jakarta.activation:1.2.2') {
        exclude group: 'jakarta.activation', module: 'jakarta.activation-api'
        exclude group: 'javax.activation', module: 'javax.activation-api'
    }
```

Reply:
As I mentioned in my original comment, I'd suggest excluding the dependency brought in by jakarta-xml.bind-api and keeping this package intact. Otherwise you might end up with incompatible api and impl

Reply:
> As I mentioned in my original comment, I'd suggest excluding the dependency brought in by jakarta-xml.bind-api and keeping this package intact. Otherwise you might end up with incompatible api and impl

```
implementation ('com.sun.activation:jakarta.activation:1.2.2') {
        exclude group: 'jakarta.activation', module: 'jakarta.activation-api'
        exclude group: 'javax.activation', module: 'javax.activation-api'
}
```
Can you verify this? Have I missed something?

Reply:
That's not quite what I meant. See my commit da73689bf498905752c592cdabf1e24300142448 as an example - or the patch below. Works at least for me. 

diff --git a/fineract-provider/dependencies.gradle b/fineract-provider/dependencies.gradle
index 4024ce12a..7a83a941c 100644
--- a/fineract-provider/dependencies.gradle
+++ b/fineract-provider/dependencies.gradle
@@ -68,9 +68,6 @@ dependencies {
             'com.github.spullara.mustache.java:compiler',
             'com.jayway.jsonpath:json-path',
 
-            // JAX-B dependencies for JDK 9+
-            'jakarta.xml.bind:jakarta.xml.bind-api',
-
             'org.dom4j:dom4j',
 
             'javax.cache:cache-api',
@@ -83,6 +80,9 @@ dependencies {
             'com.squareup.retrofit2:converter-gson',
             'com.sun.activation:jakarta.activation:1.2.2'
             )
+    implementation ('jakarta.xml.bind:jakarta.xml.bind-api') {
+        exclude group: 'jakarta.activation'
+    }
     implementation ('org.apache.activemq:activemq-broker') {
         exclude group: 'org.apache.geronimo.specs'
     }

Reply:
I've tried these changes earlier. Didn't work for me. Let me try this by clearing the gradle dependencies first :)

Reply:
Ok - what did not work when you tried it? At least it built fine for me, no issues with duplicates etc. W

Reply:
Avoid installing jakarta-api. I'm trying again by clearing all caches. Will update you ASAP

Reply:
> Ok - what did not work when you tried it? At least it built fine for me, no issues with duplicates etc.

Would you mind taking a look at this PR too? https://github.com/apache/fineract/pull/1611

Reply:
@ptuomola Thanks. It worked !!!

Reply:
@BLasan @ptuomola I've not looked closely here, but it seems to be that using             'com.sun.activation:jakarta.activation:1.2.2' is confusing, because it (probably, not verified) contains the exact same Java classes as the 'jakarta.activation' artifact... we should clean this up IMHO, but that could be done in a future PR.